src/ir/access_path_selectors_test.cc

```diff
@@ -0,0 +1,140 @@
+#include "src/ir/access_path_selectors.h"
+
+#include <string>
+#include <vector>
+
+#include "absl/hash/hash_testing.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/str_split.h"
+#include "src/common/testing/gtest.h"
+#include "src/ir/field_selector.h"
+#include "src/ir/selector.h"
+
+namespace raksha::ir {
+
+// Selector Access paths are easier to read and write in their string form, so
+// this is very useful for creating readable test inputs. This could possibly
+// be useful as a method in AccessPathSelectors itself, but, it's a bit
+// simplified right now, as it only splits on '.', the field separator
+// character. Also, I haven't found a use for it in non-test code yet.
+static absl::StatusOr<AccessPathSelectors> MakeSelectorAccessPathFromString(
+    const std::string str) {
```

Comment:
Isn't this a place where `const std::string&` is warranted to avoid a copy constructor invocation? Or, does simply having a `const` have the same effect?

Reply:
I've gotten rid of the const, which should make this more friendly to copy-elision.